Tests/MapboxMapsTests/Foundation/Mocks/MockStyleManager.swift

```diff
@@ -0,0 +1,331 @@
+import Foundation
+@testable import MapboxMaps
+@_implementationOnly import MapboxCommon_Private
+
+final class MockStyleManager: StyleManagerType, Stubbable {
+
+    func asStyleManager() -> StyleManager {
+        fatalError()
+    }
+
+    @Stubbed var stubStyleURI: String!
+    func getStyleURI() -> String {
+        stubStyleURI ?? ""
+    }
+
+    func setStyleURIForUri(_ uri: String) {
+        stubStyleURI = uri
+    }
+
+    @Stubbed var stubStyleJSON: String!
+    func getStyleJSON() -> String {
+        stubStyleJSON
+    }
+
+    func setStyleJSONForJson(_ json: String) {
+        stubStyleJSON = json
+    }
+
+    @Stubbed var stubDefaultCamera: MapboxCoreMaps.CameraOptions!
+    func getStyleDefaultCamera() -> MapboxCoreMaps.CameraOptions {
+        stubDefaultCamera
+    }
+
+    @Stubbed var stubStyleTransition: MapboxCoreMaps.TransitionOptions!
+    func getStyleTransition() -> MapboxCoreMaps.TransitionOptions {
+        stubStyleTransition
+    }
+
+    func setStyleTransitionFor(_ transitionOptions: MapboxCoreMaps.TransitionOptions) {
+        stubStyleTransition = transitionOptions
+    }
+
+    // MARK: Style Layers
+
+    private let stubStyleLayers: [MapboxCoreMaps.StyleObjectInfo] = .random(withLength: 3) {
+        MapboxCoreMaps.StyleObjectInfo(
+            id: .randomAlphanumeric(withLength: 12),
+            type: MapboxMaps.LayerType.random().rawValue)
+    }
+    func styleLayerExists(forLayerId layerId: String) -> Bool {
+        stubStyleLayers.contains(where: { $0.id == layerId })
+    }
+
+    func getStyleLayers() -> [MapboxCoreMaps.StyleObjectInfo] {
+        stubStyleLayers
+    }
+
+    func getStyleLayerProperty(
+        forLayerId layerId: String,
+        property: String
+    ) -> MapboxCoreMaps.StylePropertyValue {
+
+        fatalError()
+    }
+
+    static func getStyleLayerPropertyDefaultValue(
+        forLayerType layerType: String,
+        property: String)
+    -> MapboxCoreMaps.StylePropertyValue {
+
+        fatalError()
+    }
```

Comment:
Would we like to keep this implementation failable or we are going to fix it later?

Reply:
we are going to fix it later in this PR :-D